docs/writing-tests/making-a-testing-plan.md

```diff
@@ -0,0 +1,507 @@
+
+## Understanding the "testing surface"
+
+Web platform specifications are instructions about how a feature should work.
+They're critical for implementers to "build the right thing," but they are also
+important for anyone writing tests. We can use the same instructions to infer
+what kinds of tests would be likely to detect mistakes. Here are a few common
+patterns in specification text and the kind of tests they suggest.
+
+### Input sources
+
+Algorithms may accept input from many sources. Modifying the input is the most
+direct way we can influence the browser's behavior and verify that it matches
+the specifications. That's why it's helpful to be able to recognize all the
+different sources of input.
+
+```eval_rst
+================ ==============================================================
+Type of feature  Potential input sources
+================ ==============================================================
+JavaScript       parameters, `context object <https://dom.spec.whatwg.org/#context-object>`_
+HTML             element content, attributes, attribute values
+CSS              selector strings, property values, markup
+================ ==============================================================
+```
+
+*Example:* This is the first step of the `Notification` constructor from [the
+Notifications standard](https://notifications.spec.whatwg.org/#constructors):
+
+> The Notification(title, options) constructor, when invoked, must run these steps:
+>
+> 1. If the [current global
+>    object](https://html.spec.whatwg.org/multipage/webappapis.html#current-global-object)
+>    is a
+>    [ServiceWorkerGlobalScope](https://w3c.github.io/ServiceWorker/#serviceworkerglobalscope)
+>    object, then [throw](https://heycam.github.io/webidl/#dfn-throw) a
+>    `TypeError` exception.
+> 2. Let *notification* be the result of [creating a
+>    notification](https://notifications.spec.whatwg.org/#create-a-notification)
+>    given *title* and *options*. Rethrow any exceptions.
+>
+> [...]
+
+A thorough test suite for this constructor will include tests for the behavior
+of many different values of the *title* parameter and the *options* parameter.
+Choosing those values is a challenge unto itself--see [Avoid Excessive
+Breadth](#avoid-excessive-breadth) for advice on the topic.
+
+### Browser state
+
+The state of the browser may also influence algorithm behavior. Examples
+include the current document, the dimensions of the viewport, and the entries
+in the browsing history. Just like with direct input, a thorough set of tests
+will likely need to control these values. Browser state is often more expensive
+to manipulate (whether in terms of code, execution time, or system resources),
+and you may want to design your tests to mitigate these costs (e.g. by writing
+many subtests from the same state).
+
+*Example:* In [the `Notification` constructor referenced
+above](https://notifications.spec.whatwg.org/#constructors), the type of "the
```

Comment:
Because the text says 'referenced above', I clicked this link expecting it to take me back to the text up above. Instead it sent me to the notifications spec which was unexpected.

Reply:
If the link text said only "referenced above," then I'd understand. But because it reads "the Notification constructor referenced above," it seems clear that the target is the constructor, not the reference.